internal/handlers/cookie_credentials_handler.go

```diff
@@ -0,0 +1,31 @@
+package handlers
+
+import (
+	"errors"
+	"net/http"
+
+	"github.com/ydataai/go-core/pkg/common/logging"
+)
+
+// CookieCredentialsHandler defines a CookieCredentialsHandler struct.
+type CookieCredentialsHandler struct {
+	logger logging.Logger
+}
+
+// NewCookieCredentialsHandler defines a new CookieCredentialsHandler struct.
+func NewCookieCredentialsHandler(logger logging.Logger) CredentialsHandler {
+	return &CookieCredentialsHandler{
+		logger: logger,
+	}
+}
+
+// Extract is an interface that extracts credential information from the cookie.
+func (ac *CookieCredentialsHandler) Extract(r *http.Request) (string, error) {
+	token, err := r.Cookie("access_token")
+	if err != nil || token.Value == "" {
+		return "", errors.New("'access_token' cookie not found")
```

Comment:
instead of this, create a `notFound` error, a constant that you can put in a shared place and use it in all the handlers.

Reply:
also, log what happened

Reply:
about what happens when an error occurs,
I'm sending the error log to upstream.

please, see again and tell me if now it's ok.